tests/fault_tolerance/hardware/fault-injection-service/cuda-fault-injection/README.md

```diff
@@ -0,0 +1,178 @@
+
+- **gcc compiler** (for building the library)
+- **kubectl** with cluster access
+- Python packages: `kubernetes`, `requests`
+
+## Writing Your Own Test
+
+### 1. Build the Library (First Time Only)
+
+```bash
+cd cuda-fault-injection
+make  # Creates fake_cuda_xid79.so
+```
+
+### 2. Import Helper Functions
+
+```python
+import sys
+import os
+from pathlib import Path
+
+# Add cuda-fault-injection to path
+cuda_injection_dir = Path(__file__).parent.parent / "cuda-fault-injection"
+sys.path.insert(0, str(cuda_injection_dir))
+
+from inject_into_pods import (
+    create_cuda_fault_configmap,      # Step 1: Create ConfigMap with library
+    patch_deployment_env,             # Step 2: Patch deployment to use it
+    delete_cuda_fault_configmap       # Cleanup: Remove ConfigMap
+)
+```
+
+### 3. Use in Your Test
+
+```python
+def test_gpu_failure_xid79():
+    """Test XID 79 - GPU fell off bus"""
+    deployment_name = "vllm-worker"
+    namespace = "default"
+    target_node = "node-with-gpu"
+
+    # Enable CUDA fault injection for XID 79 (default)
+    create_cuda_fault_configmap(namespace)
+    patch_deployment_env(deployment_name, namespace, enable=True,
+                        use_configmap=True, target_node=target_node,
+                        xid_type=79)
+
+    # Pods will crash with CUDA_ERROR_NO_DEVICE
+    # ... your test logic here ...
+
+    # Cleanup
+    patch_deployment_env(deployment_name, namespace, enable=False, use_configmap=True)
+    delete_cuda_fault_configmap(namespace)
+
+def test_ecc_error_xid48():
+    """Test XID 48 - Double-bit ECC error"""
+    create_cuda_fault_configmap(namespace)
+    patch_deployment_env(deployment_name, namespace, enable=True,
+                        use_configmap=True, xid_type=48)
+
+    # Pods will crash with CUDA_ERROR_ECC_UNCORRECTABLE
+    # ... your test logic here ...
+
+    # Cleanup
+    patch_deployment_env(deployment_name, namespace, enable=False, use_configmap=True)
+    delete_cuda_fault_configmap(namespace)
+```
+
+## Key Functions
+
+### `create_cuda_fault_configmap(namespace)`
+Creates a ConfigMap containing the C source code. An init container compiles it in-pod (Linux-compatible).
+
+### `patch_deployment_env(deployment_name, namespace, enable, use_configmap, target_node, xid_type)`
+- **enable=True**: Adds LD_PRELOAD env var, mounts ConfigMap, adds init container
+- **enable=False**: Removes all CUDA fault artifacts
+- **target_node**: Pins pods to specific node (simulates real XID where failure is localized)
+- **xid_type**: XID error to simulate (79, 48, 94, 95, 43, 74). Default: 79
+
+Works with both standard Deployments and DynamoGraphDeployments.
+
+### `delete_cuda_fault_configmap(namespace)`
+Removes the ConfigMap after test.
+
+## How It Works
+
+1. **ConfigMap** stores `fake_cuda_xid79.c` source code (not compiled `.so` - ensures Linux compatibility since local build is macOS)
+2. **Init container** compiles source in-pod to `/cuda-fault/fake_cuda_xid79.so`
```

Comment:
Is line 45 above wrong then? If this is only a deployment test then why would I need to build it locally?